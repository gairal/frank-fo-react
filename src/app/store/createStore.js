import { applyMiddleware, compose, createStore } from 'redux';
import thunk from 'redux-thunk';
import { browserHistory } from 'react-router';
import reducers from './reducers';
import { updateLocation } from './location';

export default (initialState = {}) => {
  // ======================================================
  // Middleware Configuration
  // ======================================================
  const middleware = [thunk];

  // ======================================================
  // Store Instantiation and HMR Setup
  // ======================================================
  const store = createStore(
    reducers(),
    initialState,
    compose(applyMiddleware(...middleware)),
  );
  store.asyncReducers = {};

  // To unsubscribe, invoke `store.unsubscribeHistory()` anytime
  store.unsubscribeHistory = browserHistory.listen(updateLocation(store));

  return store;
};
