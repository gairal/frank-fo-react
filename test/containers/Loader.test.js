import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import createStore from '@/store/createStore';
import LoaderComponent from '@/components/core/Loader';
import LoaderContainer from '@/containers/Loader';

describe('LoaderContainer', () => {
  const store = createStore();
  let component;
  let dumbComponent;
  beforeEach(() => {
    const Container = LoaderContainer(store);
    component = renderer.create(<Provider store={store}><Container /></Provider>);
    dumbComponent = component.root.findByType(LoaderComponent);
  });

  it('Contains default props', () => {
    expect(dumbComponent.props.isFetching).toEqual(false);
  });

  it('Renders correctly', () => {
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it.skip('Can change state', () => {
    dumbComponent.props.actions.showLoader();
    // store.dispatch(() => ({ type: ' LOADER_SHOW' }));
    // expect(dumbComponent.props.isFetching).toBeTruthy();
    // store.dispatch(() => ({ type: ' LOADER_HIDE' }));
    // expect(dumbComponent.props.isFetching).toBeFalsy();
  });
});
