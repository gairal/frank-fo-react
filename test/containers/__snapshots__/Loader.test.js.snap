// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoaderContainer Renders correctly 1`] = `null`;
